### PR TITLE
dashboard: Add ping handler, allow form encoded request to login

### PR DESCRIPTION
**Pull request by jvatic**
Ping handler is accessable from any localhost address so the installer
may verify the certificate works.

Supporting form encoded request to login allows the installer to submit
a form pointing at the dashboard login endpoint (cookie will not be set
for the dashboard interface if request made via JavaScript).

### CODE REVIEW

**Comment by jvatic** [comment]
refs  #1186

File: dashboard/api.go [hunk 1]
```diff
@@ -71,19 +73,47 @@ func APIHandler(conf *Config) http.Handler {
...
 func login(req *http.Request, w http.ResponseWriter, info LoginInfo, rh RequestHelper, conf *Config) {
 	if len(info.Token) != len(conf.LoginToken) || subtle.ConstantTimeCompare([]byte(info.Token), []byte(conf.LoginToken)) != 1 {
 		rh.Error(ErrInvalidLoginToken)
 		return
 	}
 	rh.SetAuthenticated()
-	rh.WriteHeader(200)
+	if strings.Contains(req.Header.Get("Content-Type"), "form-urlencoded") {
+		w.Header().Set("Location", conf.CookiePath)
+		rh.WriteHeader(302)
```
Comment (titanous) [inline dashboard/api.go]: use `http.Redirect`

Comment (jvatic) [inline dashboard/api.go]: Done.

**Comment by titanous** [comment]
LGTM

**Pull request merged**